com.topcoder.util.config.ConfigManagerException: The property 'implementor' is null or empty.
	at com.topcoder.util.config.Helper.createObj(Helper.java:178)
	at com.topcoder.util.config.ConfigManager.getInstance(ConfigManager.java:378)
	at com.topcoder.util.log.LogFactory.getLogWithExceptions(LogFactory.java:371)
	at com.topcoder.util.log.LogFactory.getLog(LogFactory.java:294)
	at com.cronos.termsofuse.dao.impl.BaseTermsOfUseDao.<init>(BaseTermsOfUseDao.java:92)
	at com.cronos.termsofuse.dao.impl.TermsOfUseDaoImpl.<init>(TermsOfUseDaoImpl.java:250)
	at com.cronos.termsofuse.TestsHelper.loadDB(TestsHelper.java:213)
	at com.cronos.termsofuse.Demo.setUp(Demo.java:64)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:27)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:73)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:46)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:180)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:41)
	at org.junit.runners.ParentRunner$1.evaluate(ParentRunner.java:173)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:28)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:220)
	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:39)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getTermsOfUseByUserId(long userId)].
Input parameters[userId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getTermsOfUseByUserId(long userId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.Helper.getTermsOfUse(Helper.java:256)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getTermsOfUseByUserId(UserTermsOfUseDaoImpl.java:284)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_getTermsOfUseByUserId_1(UserTermsOfUseDaoImplFailureTest.java:228)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 22 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 25 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getUsersByTermsOfUseId(long termsOfUseId)].
Input parameters[termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getUsersByTermsOfUseId(long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.getUsersByTermsOfUseId(UserTermsOfUseDaoImpl.java:308)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_getUsersByTermsOfUseId_1(UserTermsOfUseDaoImplFailureTest.java:242)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 21 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 24 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasRecord(UserTermsOfUseDaoImpl.java:409)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUse(UserTermsOfUseDaoImpl.java:357)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_hasTermsOfUse_1(UserTermsOfUseDaoImplFailureTest.java:257)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 22 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 25 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasRecord(UserTermsOfUseDaoImpl.java:409)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(UserTermsOfUseDaoImpl.java:379)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_hasTermsOfUseBan_1(UserTermsOfUseDaoImplFailureTest.java:271)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 22 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 25 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasRecord(UserTermsOfUseDaoImpl.java:409)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(UserTermsOfUseDaoImpl.java:379)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(UserTermsOfUseDaoImpl.java:206)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_createUserTermsOfUse_1(UserTermsOfUseDaoImplFailureTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 23 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 26 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:10000]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:10000]
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Output parameter: false
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: A database access error has occurred.
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:317)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(UserTermsOfUseDaoImpl.java:213)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_createUserTermsOfUse_2(UserTermsOfUseDaoImplFailureTest.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: java.sql.SQLException: Missing key in referenced table for referential constraint (informix.userterms_terms_fk).
	at com.informix.jdbc.IfxSqli.a(IfxSqli.java:3204)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3518)
	at com.informix.jdbc.IfxSqli.dispatchMsg(IfxSqli.java:2353)
	at com.informix.jdbc.IfxSqli.receiveMessage(IfxSqli.java:2269)
	at com.informix.jdbc.IfxSqli.executeCommand(IfxSqli.java:775)
	at com.informix.jdbc.IfxResultSet.b(IfxResultSet.java:291)
	at com.informix.jdbc.IfxStatement.c(IfxStatement.java:1253)
	at com.informix.jdbc.IfxPreparedStatement.executeUpdate(IfxPreparedStatement.java:408)
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:315)
	... 21 more
Caused by: java.sql.SQLException: ISAM error: no record found.
	at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:373)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3523)
	... 28 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:10000, termsOfUseId:1]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:10000, termsOfUseId:1]
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Output parameter: false
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: A database access error has occurred.
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:317)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(UserTermsOfUseDaoImpl.java:213)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_createUserTermsOfUse_6(UserTermsOfUseDaoImplFailureTest.java:150)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: java.sql.SQLException: Missing key in referenced table for referential constraint (informix.usertermsofusexref_user_fk).
	at com.informix.jdbc.IfxSqli.a(IfxSqli.java:3204)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3518)
	at com.informix.jdbc.IfxSqli.dispatchMsg(IfxSqli.java:2353)
	at com.informix.jdbc.IfxSqli.receiveMessage(IfxSqli.java:2269)
	at com.informix.jdbc.IfxSqli.executeCommand(IfxSqli.java:775)
	at com.informix.jdbc.IfxResultSet.b(IfxResultSet.java:291)
	at com.informix.jdbc.IfxStatement.c(IfxStatement.java:1253)
	at com.informix.jdbc.IfxPreparedStatement.executeUpdate(IfxPreparedStatement.java:408)
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:315)
	... 21 more
Caused by: java.sql.SQLException: ISAM error: no record found.
	at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:373)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3523)
	... 28 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Output parameter: false
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:1, termsOfUseId:1]
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Output parameter: false
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: A database access error has occurred.
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:317)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(UserTermsOfUseDaoImpl.java:213)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_createUserTermsOfUse_3(UserTermsOfUseDaoImplFailureTest.java:163)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: java.sql.SQLException: Unique constraint (informix.u116_29) violated.
	at com.informix.jdbc.IfxSqli.a(IfxSqli.java:3204)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3518)
	at com.informix.jdbc.IfxSqli.dispatchMsg(IfxSqli.java:2353)
	at com.informix.jdbc.IfxSqli.receiveMessage(IfxSqli.java:2269)
	at com.informix.jdbc.IfxSqli.executeCommand(IfxSqli.java:775)
	at com.informix.jdbc.IfxResultSet.b(IfxResultSet.java:291)
	at com.informix.jdbc.IfxStatement.c(IfxStatement.java:1253)
	at com.informix.jdbc.IfxPreparedStatement.executeUpdate(IfxPreparedStatement.java:408)
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:315)
	... 21 more
Caused by: java.sql.SQLException: ISAM error: duplicate value for a record with unique key.
	at com.informix.util.IfxErrMsg.getSQLException(IfxErrMsg.java:373)
	at com.informix.jdbc.IfxSqli.E(IfxSqli.java:3523)
	... 28 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:2, termsOfUseId:1]
Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Input parameters[userId:2, termsOfUseId:1]
Exiting method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.hasTermsOfUseBan(long userId, long termsOfUseId)].
Output parameter: true
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.UserBannedException: The user with id 2 has a ban for following terms of use with id 1.
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.createUserTermsOfUse(UserTermsOfUseDaoImpl.java:208)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_createUserTermsOfUse_4(UserTermsOfUseDaoImplFailureTest.java:175)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:2, termsOfUseId:2]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.TermsOfUsePersistenceException: Failed to create a connection.
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:195)
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:308)
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:346)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(UserTermsOfUseDaoImpl.java:248)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_removeUserTermsOfUse_1(UserTermsOfUseDaoImplFailureTest.java:190)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)
Caused by: com.topcoder.db.connectionfactory.DBConnectionException: error occurs while creating the connection.
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:450)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:406)
	at com.topcoder.db.connectionfactory.DBConnectionFactoryImpl.createConnection(DBConnectionFactoryImpl.java:477)
	at com.cronos.termsofuse.dao.impl.Helper.createConnection(Helper.java:193)
	... 23 more
Caused by: java.sql.SQLException: Incorrect password or user com.informix.asf.IfxASFRemoteException: informix@pc001 is not known on the database server.
	at com.informix.jdbc.IfxSqliConnect.<init>(IfxSqliConnect.java:964)
	at sun.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
	at com.informix.jdbc.IfxDriver.connect(IfxDriver.java:254)
	at java.sql.DriverManager.getConnection(DriverManager.java:582)
	at java.sql.DriverManager.getConnection(DriverManager.java:154)
	at com.topcoder.db.connectionfactory.producers.JDBCConnectionProducer.createConnection(JDBCConnectionProducer.java:448)
	... 26 more

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:2, termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.EntityNotFoundException: The entity was not found for id (2, 1).
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:348)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(UserTermsOfUseDaoImpl.java:248)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_removeUserTermsOfUse_2(UserTermsOfUseDaoImplFailureTest.java:202)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)

Entering method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
Input parameters[userId:100000, termsOfUseId:1]
Error in method [com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(long userId, long termsOfUseId)].
com.cronos.termsofuse.dao.EntityNotFoundException: The entity was not found for id (100000, 1).
	at com.cronos.termsofuse.dao.impl.Helper.executeUpdate(Helper.java:348)
	at com.cronos.termsofuse.dao.impl.UserTermsOfUseDaoImpl.removeUserTermsOfUse(UserTermsOfUseDaoImpl.java:248)
	at com.cronos.termsofuse.dao.failure.UserTermsOfUseDaoImplFailureTest.test_removeUserTermsOfUse_3(UserTermsOfUseDaoImplFailureTest.java:214)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at junit.framework.TestCase.runTest(TestCase.java:168)
	at junit.framework.TestCase.runBare(TestCase.java:134)
	at junit.framework.TestResult$1.protect(TestResult.java:110)
	at junit.framework.TestResult.runProtected(TestResult.java:128)
	at junit.framework.TestResult.run(TestResult.java:113)
	at junit.framework.TestCase.run(TestCase.java:124)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at junit.framework.TestSuite.runTest(TestSuite.java:232)
	at junit.framework.TestSuite.run(TestSuite.java:227)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:420)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:911)
	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:768)


